app/helpers/application_helper.rb

```diff
@@ -2,17 +2,17 @@
 # frozen_string_literal: true
 module ApplicationHelper
   def admin?
-    return true if current_user && current_user.admin?
+    return true if user_signed_in? && current_user.admin?
```

Comment:
Now look what you have:
```current_user``` will always return instance of ```User``` class.
I user is not signed in than ```current_user``` will return empty instance of ```User``` class. 
Call of ```#admin?``` or ```#moderator?``` on empty instance of ```User``` class will always return false.
So yo can safely remove ```user_signed_in?``` from this conditions.